Fix DieHard build

It recently [moved to cmake](https://github.com/emeryberger/DieHard?tab=readme-ov-file#linux--mac):

```
 --------------------------------------------
build dh: version master
--------------------------------------------

~/work/mimalloc-bench/mimalloc-bench/extern ~/work/mimalloc-bench/mimalloc-bench
Cloning into 'dh'...
Already on 'master'
Your branch is up to date with 'origin/master'.
make: *** No rule to make target 'linux-gcc-64'.  Stop.
make: Entering directory '/home/runner/work/mimalloc-bench/mimalloc-bench/extern/dh/src'
make: Leaving directory '/home/runner/work/mimalloc-bench/mimalloc-bench/extern/dh/src'
Error: Process completed with exit code 2.
```

--- a/bench.sh
+++ b/bench.sh
@@ -97,7 +97,7 @@ readonly lib_tbb="$localdevdir/tbb/bench_release/libtbbmalloc_proxy$extso"
 readonly lib_tbb_dir="$(dirname $lib_tbb)"
 
 
-alloc_lib_add "dh"     "$localdevdir/dh/src/libdieharder$extso"
+alloc_lib_add "dh"     "$localdevdir/dh/src/build/libdieharder$extso"
 alloc_lib_add "ff"     "$localdevdir/ff/libffmallocnpmt$extso"
 alloc_lib_add "fg"     "$localdevdir/fg/libfreeguard$extso"
 alloc_lib_add "gd"     "$localdevdir/gd/libguarder$extso"

--- a/build-bench-env.sh
+++ b/build-bench-env.sh
@@ -566,10 +566,8 @@ if test "$setup_dh" = "1"; then
   # remove all the historical useless junk
   rm -rf ./benchmarks/ ./src/archipelago/ ./src/build/ ./src/exterminator/ ./src/local/ ./src/original-diehard/ ./src/replicated/ ./docs
   cd src
-  mkdir -p build
-  cd build
-  cmake ..
-  make
+  cmake -S . -B build
+  cmake --build build -j `nproc`
   cd ../..
   popd
 fi

--- a/build-bench-env.sh
+++ b/build-bench-env.sh
@@ -37,7 +37,7 @@ rebuild=0
 all=0
 
 # allocator versions
-readonly version_dh=640949fe0128d9c7013677c8c332698d5c2cefc2   # last unmaintained version, see https://github.com/daanx/mimalloc-bench/issues/222 for the up-to-date one
+readonly version_dh=master
 readonly version_ff=master   # ~unmaintained since 2021
 readonly version_fg=master   # ~unmaintained since 2018
 readonly version_gd=master   # ~unmaintained since 2021
@@ -565,11 +565,12 @@ if test "$setup_dh" = "1"; then
   checkout dh $version_dh https://github.com/emeryberger/DieHard
   # remove all the historical useless junk
   rm -rf ./benchmarks/ ./src/archipelago/ ./src/build/ ./src/exterminator/ ./src/local/ ./src/original-diehard/ ./src/replicated/ ./docs
-  if test "$darwin" = "1"; then
-    TARGET=libdieharder make -C src macos
-  else
-    TARGET=libdieharder make -C src linux-gcc-64
-  fi
+  cd src
+  mkdir -p build
+  cd build
+  cmake ..
+  make
+  cd ../..
   popd
 fi
 

